From 4b41a7d182e72b920ebe098778f83afc80042e59 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 3 Oct 2023 12:04:54 +0530
Subject: [PATCH] Fix #6650

---
 glfw/dbus_glfw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glfw/dbus_glfw.c b/glfw/dbus_glfw.c
index a4fbdf0eb2..90090a146d 100644
--- a/glfw/dbus_glfw.c
+++ b/glfw/dbus_glfw.c
@@ -33,12 +33,12 @@
 
 static void
 report_error(DBusError *err, const char *fmt, ...) {
-    static char buf[1024];
+    static char buf[4096];
     va_list args;
     va_start(args, fmt);
     int n = vsnprintf(buf, sizeof(buf), fmt, args);
     va_end(args);
-    snprintf(buf + n, sizeof(buf), ". DBUS error: %s", err->message);
+    if ((size_t)n < (sizeof(buf) - 256)) snprintf(buf + n, sizeof(buf) - n, ". DBUS error: %s", err->message ? err->message : "(null)");
     _glfwInputError(GLFW_PLATFORM_ERROR, "%s", buf);
     dbus_error_free(err);
 }
