From d9cd92d4ed5854631d904665abffdec94cbdb27d Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 3 Oct 2023 12:07:19 +0530
Subject: [PATCH] ...

---
 glfw/dbus_glfw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glfw/dbus_glfw.c b/glfw/dbus_glfw.c
index 90090a146d..252e8ac698 100644
--- a/glfw/dbus_glfw.c
+++ b/glfw/dbus_glfw.c
@@ -38,7 +38,7 @@ report_error(DBusError *err, const char *fmt, ...) {
     va_start(args, fmt);
     int n = vsnprintf(buf, sizeof(buf), fmt, args);
     va_end(args);
-    if ((size_t)n < (sizeof(buf) - 256)) snprintf(buf + n, sizeof(buf) - n, ". DBUS error: %s", err->message ? err->message : "(null)");
+    if (n >= 0 && (size_t)n < (sizeof(buf) - 256)) snprintf(buf + n, sizeof(buf) - n, ". DBUS error: %s", err->message ? err->message : "(null)");
     _glfwInputError(GLFW_PLATFORM_ERROR, "%s", buf);
     dbus_error_free(err);
 }
