diff --git a/setup.py b/setup.py
index 5d866c993..3174fc47f 100755
--- a/setup.py
+++ b/setup.py
@@ -970,6 +970,7 @@ def build_static_kittens(
         raise SystemExit(f'The version of go on this system ({current_go_version}) is too old. go >= {required_go_version} is needed')
     if not for_platform:
         update_go_generated_files(args, os.path.join(launcher_dir, appname))
+    return
     cmd = [go, 'build', '-v']
     vcs_rev = args.vcs_rev or get_vcs_rev()
     ld_flags = [f"-X 'kitty.VCSRevision={vcs_rev}'"]
