diff --git a/gen-go-code.py b/gen-go-code.py
index 25b902fa1..fc54b722b 100755
--- a/gen-go-code.py
+++ b/gen-go-code.py
@@ -12,6 +12,7 @@
 import subprocess
 import sys
 import tarfile
+import shutil
 from contextlib import contextmanager, suppress
 from functools import lru_cache
 from itertools import chain
@@ -851,6 +852,7 @@ def main() -> None:
             generate_unicode_names(src, dest)
     generate_ssh_kitten_data()
 
+    shutil.copy('constants_generated.go','src/kitty')
     update_completion()
     update_at_commands()
     kitten_clis()
diff --git a/setup.py b/setup.py
index 5d866c993..3174fc47f 100755
--- a/setup.py
+++ b/setup.py
@@ -970,6 +970,7 @@ def build_static_kittens(
         raise SystemExit(f'The version of go on this system ({current_go_version}) is too old. go >= {required_go_version} is needed')
     if not for_platform:
         update_go_generated_files(args, os.path.join(launcher_dir, appname))
+    return
     cmd = [go, 'build', '-v']
     vcs_rev = args.vcs_rev or get_vcs_rev()
     ld_flags = [f"-X 'kitty.VCSRevision={vcs_rev}'"]
