diff --git a/hyprbars/barDeco.cpp b/hyprbars/barDeco.cpp
index fdba993..6c38e2c 100644
--- a/hyprbars/barDeco.cpp
+++ b/hyprbars/barDeco.cpp
@@ -1,7 +1,7 @@
 #include "barDeco.hpp"
 
-#include <src/Compositor.hpp>
-#include <src/Window.hpp>
+#include <hyprland/src/Compositor.hpp>
+#include <hyprland/src/Window.hpp>
 
 #include "globals.hpp"
 
@@ -330,4 +330,4 @@ SWindowDecorationExtents CHyprBar::getWindowDecorationReservedArea() {
 Vector2D CHyprBar::cursorRelativeToBar() {
     static auto* const PHEIGHT = &HyprlandAPI::getConfigValue(PHANDLE, "plugin:hyprbars:bar_height")->intValue;
     return g_pInputManager->getMouseCoordsInternal() - m_pWindow->m_vRealPosition.vec() + Vector2D{0, *PHEIGHT};
-}
\ No newline at end of file
+}
diff --git a/hyprbars/barDeco.hpp b/hyprbars/barDeco.hpp
index ef59508..69e0147 100644
--- a/hyprbars/barDeco.hpp
+++ b/hyprbars/barDeco.hpp
@@ -2,8 +2,8 @@
 
 #define WLR_USE_UNSTABLE
 
-#include <src/render/decorations/IHyprWindowDecoration.hpp>
-#include <src/render/OpenGL.hpp>
+#include <hyprland/src/render/decorations/IHyprWindowDecoration.hpp>
+#include <hyprland/src/render/OpenGL.hpp>
 #include "globals.hpp"
 
 class CHyprBar : public IHyprWindowDecoration {
@@ -52,4 +52,4 @@ class CHyprBar : public IHyprWindowDecoration {
 
     bool                     m_bDraggingThis = false;
     bool                     m_bDragPending  = false;
-};
\ No newline at end of file
+};
diff --git a/hyprbars/globals.hpp b/hyprbars/globals.hpp
index 37e8363..66c4f04 100644
--- a/hyprbars/globals.hpp
+++ b/hyprbars/globals.hpp
@@ -1,5 +1,5 @@
 #pragma once
 
-#include <src/plugins/PluginAPI.hpp>
+#include <hyprland/src/plugins/PluginAPI.hpp>
 
-inline HANDLE PHANDLE = nullptr;
\ No newline at end of file
+inline HANDLE PHANDLE = nullptr;
diff --git a/hyprbars/main.cpp b/hyprbars/main.cpp
index 2250ef6..24c8cda 100644
--- a/hyprbars/main.cpp
+++ b/hyprbars/main.cpp
@@ -3,9 +3,9 @@
 #include <unistd.h>
 
 #include <any>
-#include <src/Compositor.hpp>
-#include <src/Window.hpp>
-#include <src/config/ConfigManager.hpp>
+#include <hyprland/src/Compositor.hpp>
+#include <hyprland/src/Window.hpp>
+#include <hyprland/src/config/ConfigManager.hpp>
 
 #include "barDeco.hpp"
 #include "globals.hpp"
@@ -51,4 +51,4 @@ APICALL EXPORT PLUGIN_DESCRIPTION_INFO PLUGIN_INIT(HANDLE handle) {
 APICALL EXPORT void PLUGIN_EXIT() {
     for (auto& m : g_pCompositor->m_vMonitors)
         m->scheduledRecalc = true;
-}
\ No newline at end of file
+}
