diff --git a/hyprtrails/globals.hpp b/hyprtrails/globals.hpp
index f5d8802..5abe391 100644
--- a/hyprtrails/globals.hpp
+++ b/hyprtrails/globals.hpp
@@ -5,8 +5,8 @@
 inline HANDLE PHANDLE = nullptr;
 
 struct SGlobalState {
-    CShader          trailShader;
+    SShader          trailShader;
     wl_event_source* tick = nullptr;
 };
 
-inline UP<SGlobalState> g_pGlobalState;
\ No newline at end of file
+inline UP<SGlobalState> g_pGlobalState;
