diff --git a/CMakeLists.txt b/CMakeLists.txt
index c6582ca..e786159 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,5 +44,5 @@ add_dependencies(tests hyprlang_fuzz)
 # Installation
 install(TARGETS hyprlang
         PUBLIC_HEADER DESTINATION include
-        LIBRARY DESTINATION lib)
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${CMAKE_BINARY_DIR}/hyprlang.pc DESTINATION ${PREFIX}/share/pkgconfig)
