diff --git a/CMakeLists.txt b/CMakeLists.txt
index be38f90..a229cdf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,8 +70,5 @@ add_test(NAME "Test libhyprcursor in C" WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/te
 add_dependencies(tests hyprcursor_test_c)
 
 # Installation
-install(DIRECTORY "include/hyprcursor" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} DIRECTORY_PERMISSIONS 
-    OWNER_WRITE OWNER_READ OWNER_EXECUTE
-    GROUP_WRITE GROUP_READ GROUP_EXECUTE
-    WORLD_WRITE WORLD_READ WORLD_EXECUTE)
+install(DIRECTORY "include/hyprcursor" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(FILES ${CMAKE_BINARY_DIR}/hyprcursor.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
